server/src/Auth.hs

```diff
@@ -222,6 +220,53 @@ setSessionCookie token = setCookie
 -- | presignS3URL
 --------------------------------------------------------------------------------
 
+{-@ ignore photoPost @-}
+photoPost :: T.Text -> Controller ()
+photoPost id = do
+  image   <- photoImage
+  photoMb <- selectFirst (photoHash' ==. id)
+  case photoMb of
+    Just _  -> updatePhoto id image
+    Nothing -> insertPhoto id image 
+  respondJSON status200 $ "saved photo for: " <> id
+
+photoImage :: Controller (FileInfo L.ByteString)
+photoImage = do 
+  (_, files) <- decodeFiles
+  case files of
+    ("image", f) : _ -> return f
+    _                -> respondError status401 (Just "Invalid photo POST")
+
+updatePhoto :: T.Text -> FileInfo L.ByteString -> Controller ()
+updatePhoto id image = do 
+  -- logT Log.INFO ("updatePhoto: id " ++ show id ++ " " ++ show (fileName image))
+  updateWhere (photoHash' ==. id)
+    ((photoFileName'    `assign` (fileName        image)) `combine` 
+     (photoFileType'    `assign` (fileContentType image)) `combine` 
+     (photoFileContent' `assign` L.toStrict (fileContent image))
+    )
+
+insertPhoto :: T.Text -> FileInfo L.ByteString -> Controller () 
+insertPhoto id image = do 
+  let name  = fileName image
+  let typ   = fileContentType image
+  let blob  = L.toStrict (fileContent image)
+  let photo = mkPhoto id name typ blob 
+  key      <- insert photo
+  return ()
+  -- logT Log.INFO ("insertPhoto: id " ++ show (id, fileName image, key))
+
+{-@ ignore photoGet @-}
+photoGet :: T.Text -> Controller ()
+photoGet id = do
```

Comment:
Maybe we should at least require you to be logged in to see a photo.

Reply:
done.